src/routes/InstancesPage/CreateInstanceModal.js

```diff
@@ -52,26 +71,24 @@ function CreateInstanceModal({ isOpen, onClose, onRequestCreate }) {
     }));
   }
 
-  function onInputChange(value) {
+  function onChangeAWSAccountNumber(id, selection) {
     setFormValues((prevFormValues) => ({
       ...prevFormValues,
-      name: value,
+      aws_account_number: selection,
     }));
   }
 
-  async function onRequestCreateHandler() {
-    setIsRequestingCreate(true);
-    const result = await onRequestCreate(formValues);
-    setIsRequestingCreate(false);
-    if (result instanceof Error) {
-      const errorMessage = result.response.data.reason;
-      setErrorMessage(errorMessage);
-    } else {
-      setFormValues(defaultFormValues);
-      onClose();
-    }
+  function onInputChangeHandler(fieldName) {
+    return function onInputChange(value) {
+      setFormValues((prevFormValues) => ({
+        ...prevFormValues,
+        [fieldName]: value,
+      }));
+    };
   }
 
+  const onNameChange = onInputChangeHandler('name');
```

Comment:
Why the wrapper functions like this?

For text inputs in our ACS we've been able to utilize a more generic handler function, like this example:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/ArtifactoryIntegrationForm.tsx#L105-L107

Reply:
When I sent out this PR a couple of months ago, we had multiple text inputs and I did this because there were similarities with the text change handlers. Now that a lot of changes happened, I guess it's not necessary anymore so we can remove it